Sources/MongoSwift/MongoSwift.swift

```diff
@@ -3,12 +3,34 @@ import CLibMongoC
 /// MongoSwift only supports MongoDB 3.6+.
 internal let MIN_SUPPORTED_WIRE_VERSION = 6
 
+/// Store optionally provided metadata about a library wrapping the driver.
+private var clientMetadataLibraryName: String? 
+private var clientMetadataLibraryVersion: String?
+
+/// Adds metadata to include the in the handshake performed with the MongoDB server. This is intended for use by
+/// libraries wrapping the driver e.g. MongoDBVapor or an ORM. If used, this method should be called exactly once.
+/// This method will only have an effect if called before any `MongoClient`s are initialized.
+public func addWrappingLibraryMetadata(name: String, version: String) {
```

Comment:
the API restrictions are a bit annoying, but libmongoc only allows you to call the underlying method once and only before the driver performs any handshakes